activation/activation.go

```diff
@@ -301,6 +322,43 @@ func (b *Builder) generateProof() error {
 	return nil
 }
 
+func (b *Builder) waitForFirstATX(ctx context.Context) {
+	currEpoch := b.layerClock.GetCurrentLayer().GetEpoch()
+	if currEpoch == 0 { // genesis miner
+		return
+	}
+	if prev, err := b.cdb.GetPrevAtx(b.nodeID); err == nil {
+		if prev.PubLayerID.GetEpoch() == currEpoch-1 {
```

Comment:
it should have atx that targets current epoch

Reply:
you are right. i came to the same conclusion after a good night sleep :P